src/client/pages/Geo/GeoMap/GeoMapMenuData/MapVisualizerPanel/mapVisualizerPanel.tsx

```diff
@@ -1,35 +1,103 @@
+  { key: 'ESAWorldCover', title: 'ESA (2020)', apiUri: '/api/geo/layers/forest/FIN/ESAWorldCover/' },
+  { key: 'Hansen', title: 'Hansen GFC (2020)', apiUri: '/api/geo/layers/forest/FIN/Hansen/10/' },
 ]
 
-const SatelliteSourcePanel: React.FC = () => {
+const queryOverlayLayer = (mapLayerId: string, overlayLayers: google.maps.MVCArray) => {
+  for (let i = 0; i < overlayLayers.getLength(); i += 1) {
+    const overlayLayer = overlayLayers.getAt(i)
+    if (overlayLayer.name === mapLayerId) {
+      return true
+    }
+  }
+  return false
+}
+
+const removeOverlayLayer = (mapLayerId: string, overlayLayers: google.maps.MVCArray) => {
+  for (let i = 0; i < overlayLayers.getLength(); i += 1) {
+    const overlayLayer = overlayLayers.getAt(i)
+    if (overlayLayer.name === mapLayerId) {
+      overlayLayers.removeAt(i)
+
+      break
+    }
+  }
+}
+
+const addOVerlayLayer = (map: google.maps.Map, mapId: string, mapLayerKey: string) => {
+  const tileSource = new ee.layers.EarthEngineTileSource({
+    mapid: mapId,
+  })
+  const overlay = new ee.layers.ImageOverlay(tileSource, { name: mapLayerKey })
+  map.overlayMapTypes.push(overlay)
+}
+
+const MapVisualizerPanel: React.FC = () => {
+  const dispatch = useAppDispatch()
+  const forestOptions = useForestSourceOptions()
+
+  const map = useGeoMap()
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const onCheckboxClick = useCallback(
+    async (apiUri: string, mapLayerKey: string, source: ForestSources) => {
+      const forestOptionsCopy = { ...forestOptions }
+      const sources = [...forestOptions.sources]
+
+      if (queryOverlayLayer(mapLayerKey, map.overlayMapTypes)) {
+        forestOptionsCopy.sources = sources.filter((el: ForestSources) => el !== source)
+        if (!forestOptionsCopy.sources.includes(source)) {
+          removeOverlayLayer(mapLayerKey, map.overlayMapTypes)
+        }
+      } else {
+        await axios.get(apiUri).then((response) => {
+          addOVerlayLayer(map, response.data.mapId, mapLayerKey)
+          sources.push(source)
+          forestOptionsCopy.sources = sources
+        })
+      }
+      dispatch(GeoActions.updateForestOptions(forestOptionsCopy))
+    },
+    [dispatch, forestOptions, map]
+  )
+
   return (
     <div className="geo-map-menu-data-visualizer-panel">
-      <button type="button" className="btn geo-map-menu-data-btn-upload">
-        Upload Map
-      </button>
-      <p>Choose Layers</p>
+      <p>Forest Layers</p>
       <div className="geo-map-menu-data-visualizer-panel-layers">
         {layers.map((layer, index) => (
           <GeoMapMenuListElement
-            key={layer.title}
+            key={layer.key}
             title={layer.title}
             tabIndex={index * -1 - 1}
-            checked={false}
-            onCheckboxClick={() => null}
+            checked={forestOptions.sources.includes(layer.key)}
+            onCheckboxClick={() => onCheckboxClick(layer.apiUri, layer.key, layer.key)}
```

Comment:
you could even define a type layer 
```type Layer = {apiUri:string,key:string,type:string}
const layers:Array<Layer> = ...

// and define 
const onCheckboxClick = useCallback(
     async (layer:Layer) => {
```

Reply:
Yes, actually I started creating a type Layer once I noticed I'll make more use of it, for opacity for instance, and now the palette and who knows. :)